src/FlatSharp.Runtime/NullCheckingFacadeConverter.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * Copyright 2021 James Courtney
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace FlatSharp.Runtime
+{
+    using System;
+    using System.ComponentModel;
+
+    /// <summary>
+    /// Adds a null check to Facade converters for underlying reference types.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public struct NullCheckingFacadeTypeConverter<TUnderlyingType, TType, TConverter> : IFacadeTypeConverter<TUnderlyingType, TType>
+        where TConverter : struct, IFacadeTypeConverter<TUnderlyingType, TType>
+    {
+        public TUnderlyingType ConvertToUnderlyingType(TType item)
+        {
+            var underlyingValue = default(TConverter).ConvertToUnderlyingType(item);
+            if (object.ReferenceEquals(underlyingValue, null))
```

Comment:
I believe that checking `if (underlayingValue is null)` is now the recommended way for checking null-references (since C# 7)

- https://github.com/dotnet/docs/issues/6855

Reply:
This is good feedback -- this had gone over my head at some point. I definitely like this better than using ReferenceEquals everywhere. Thanks!